corehq/form_processor/backends/couch.py

```diff
@@ -0,0 +1,78 @@
+from couchdbkit import ResourceNotFound
+
+from couchforms.models import doc_types, XFormInstance, XFormError
+from couchforms.exceptions import UnexpectedDeletedXForm
+from dimagi.utils.couch.undo import DELETED_SUFFIX
+
+from ..exceptions import XFormNotFound
+
+
+class XFormCouch(object):
+
+    @staticmethod
+    def create_from_generic(generic_xform, generic_attachment=None):
+        xform = XFormInstance.from_generic(generic_xform)
+        xform.save()
+        if generic_attachment:
+            xform.put_attachment(**generic_attachment.to_json())
+        return xform
+
+    @staticmethod
+    def get_attachment(xform_id, attachment_name):
+        return XFormInstance.get_db().fetch_attachment(xform_id, attachment_name)
+
+    @classmethod
+    def get_attachments(cls, xform_id):
```

Comment:
Side note, I think we can get rid of this completely. The places its being used can just use the form returned from `submit_form_locally`